apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md

```diff
@@ -0,0 +1,202 @@
+:::
+
+This script will accept `.`, if you want to add the project to the root of a folder you've already created. Otherwise, name your project. Select each option in the generation script as you see fit. We recommend the following selections:
+
+- Use Typescript?: Yes
+- Use ESLint?: Yes
+- Use Tailwind?: Your preference
+- Use `src/` directory?: Yes
+- Use App Router?: **No**
+- Customize the default import alias?: No
+
+:::caution
+For now, we recommend that you **DO NOT** use App Router. If you do, you'll need to debug some additional config and dependency issues that are out of the scope of this guide.
+:::
+
+:::info
+
+The default Next.js script installs [Tailwind]. [RainbowKit]'s does not.
+
+:::
+
+Run your app with `yarn dev` to make sure it generated correctly.
+
+### Manually Installing RainbowKit, Wagmi, and Viem
+
+The [quick start] guide for RainbowKit also contains step-by-step instructions for manual install. We'll be following an adjusted version here. Most of the setup is actually for configuring [wagmi], which sits on top of [viem] and makes it much easier to write React that interacts with the blockchain.
+
+Start by installing the dependencies. Once again, we'll use specific versions to ensure compatibility. Update your `package.json` dependencies to include:
+
+```json
+    "@rainbow-me/rainbowkit": "^1.3.0",
+    "viem": "^1.19.7",
+    "wagmi": "^1.4.5"
+```
+
+Then run:
+
+```bash
+yarn install
+```
+
+:::info
+Onchain libraries and packages tend to require very current versions of Node. If you're not already using it, you may want to install [nvm].
+:::
+
+## Adding Imports, Connectors, Config
+
+In Next.js 13 with the pages router, the root of your app is found in `src/pages/_app_.tsx`, if you followed the recommended setup options. As we want the blockchain provider context to be available for the entire app, we'll add it here.
+
+### Imports
+
+Start with the imports:
+
+```typescript
+import '@rainbow-me/rainbowkit/styles.css';
+import { getDefaultWallets, RainbowKitProvider } from '@rainbow-me/rainbowkit';
+import { configureChains, createConfig, WagmiConfig } from 'wagmi';
+import { mainnet, base, baseGoerli } from 'wagmi/chains';
+import { publicProvider } from 'wagmi/providers/public';
+```
+
+:::caution
+If you're adapting this guide to a different set of libraries or platforms, you may need to import `styles.css` differently. You'll know this is the case if you get ugly text at the bottom of the page instead of a nice modal when you click the connect button.
+:::
+
+### Connectors
+
+Next, configure the chains, wallet connectors, and providers for your app. We'll use the `publicProvider` for now, to get started. See our guide on [Connecting to the Blockchain] for more information on blockchain providers.
```

Comment:
Btw, when you use the word `Next`, I wasn't sure if you were refering to Next.js or just saying the word `next`. lol

Reply:
Haha, i'll pick a different one.